irohad/ametsuchi/impl/postgres_wsv_command.cpp

```diff
@@ -97,91 +162,184 @@ namespace iroha {
+                              "json_data: %s")
+                % account.account_id % account.domain_id % account.quorum
+                % default_tx_counter % account.json_data)
+            .str();
+      };
+
+      return makeCommandResult(result, message_gen);
     }
 
-    bool PostgresWsvCommand::insertAsset(const model::Asset &asset) {
+    WsvCommandResult PostgresWsvCommand::insertAsset(
+        const model::Asset &asset) {
       uint32_t precision = asset.precision;
-      return execute(
+      auto result = execute_(
           "INSERT INTO asset(asset_id, domain_id, \"precision\", data) "
           "VALUES ("
           + transaction_.quote(asset.asset_id) + ", "
           + transaction_.quote(asset.domain_id) + ", "
           + transaction_.quote(precision) + ", " + /*asset.data*/ "NULL"
           + ");");
+
+      auto message_gen = [&] {
+        return (boost::format("failed to insert asset, asset id: '%s', "
+                              "domain id: '%s', precision: %d")
+                % asset.asset_id % asset.domain_id % precision)
+            .str();
+      };
+
+      return makeCommandResult(result, message_gen);
     }
 
-    bool PostgresWsvCommand::upsertAccountAsset(
+    WsvCommandResult PostgresWsvCommand::upsertAccountAsset(
         const model::AccountAsset &asset) {
-      return execute(
+      auto result = execute_(
             "INSERT INTO account_has_asset(account_id, asset_id, amount) "
             "VALUES ("
             + transaction_.quote(asset.account_id) + ", "
             + transaction_.quote(asset.asset_id) + ", "
             + transaction_.quote(asset.balance.to_string())
             + ") ON CONFLICT (account_id, asset_id) DO UPDATE SET "
             "amount = EXCLUDED.amount;");
+
+      auto message_gen = [&] {
+        return (boost::format("failed to upsert account, account id: '%s', "
+                              "asset id: '%s', balance: %s")
+                % asset.account_id % asset.asset_id % asset.balance.to_string())
+            .str();
+      };
+
+      return makeCommandResult(result, message_gen);
     }
 
-    bool PostgresWsvCommand::insertSignatory(const pubkey_t &signatory) {
-      return execute("INSERT INTO signatory(public_key) VALUES ("
-                     + transaction_.quote(pqxx::binarystring(signatory.data(),
-                                                             signatory.size()))
-                     + ") ON CONFLICT DO NOTHING;");
+    WsvCommandResult PostgresWsvCommand::insertSignatory(
+        const pubkey_t &signatory) {
+      auto result = execute_("INSERT INTO signatory(public_key) VALUES ("
+                             + transaction_.quote(pqxx::binarystring(
+                                   signatory.data(), signatory.size()))
+                             + ") ON CONFLICT DO NOTHING;");
+
+      auto message_gen = [&] {
+        return (boost::format("failed to insert signatory, hex string: '%s'")
```

Comment:
Hex string of what? May be non obvious for end-user

Reply:
Not really sure that format should be written as well, imo just `"failed to insert signatory, signatory: '%s'"` is better and more consistent

Reply:
if we have failed to insert signatory, I think it is clear that hex string is the signatory.

Reply:
Maybe we can make it "signatory hex string". I think hex string is crucial, so people do not confuse it with just string